scripts/update_dep.sh

```diff
@@ -13,21 +13,51 @@ set -euo pipefail
 
 source ./scripts/test_lib.sh
 
+if [ "$#" -ne 2 ]; then
+    echo "Illegal number of parameters"
+    exit 1
+fi
+
 mod="$1"
 ver="$2"
 
-function maybe_update_module {
+function print_current_dep_version {
+  echo "${mod} version in all go mod files"
+  grep --exclude-dir=.git --include=\*.mod -Ri "${mod}" | grep -v sum
+  printf "\n\n"
+}
+
+function is_fully_indirect {
+  print_current_dep_version
+
+  # check if all lines end with "// indirect"
+  # if grep found nothing, the error code will be non-zero
+  if grep --exclude-dir=.git --include=\*.mod -Ri -q "^.*${mod} v.*// indirect$"; then 
+      echo "Fully indirect, we will terminate the script"
+      exit 1
```

Comment:
Sometimes we still need bump pure indirect dependency, i.e. due to CVE. A couple of approaches:
- raise a question something "XXX is a pure indirect dependency, Are you sure you want to proceed? (y/n):"
- Or we can just print a warning and automatically continue to execute the script. As mentioned in previous comment, it's up to maintainers/contributors whether to bump a pure indirect dependency. If not, then they shouldn't run this script at all.